sklearn/decomposition/tests/test_pca.py

```diff
@@ -531,59 +376,42 @@ def test_infer_dim_3():
     pca = PCA(n_components=p, svd_solver='full')
     pca.fit(X)
     spect = pca.explained_variance_
-    assert_greater(_infer_dimension_(spect, n, p), 2)
+    assert _infer_dimension_(spect, n, p) > 2
 
 
-def test_infer_dim_by_explained_variance():
-    X = iris.data
-    pca = PCA(n_components=0.95, svd_solver='full')
+@pytest.mark.parametrize(
+    "X, n_components, n_components_",
```

Comment:
Maybe call the first one `n_components_fraction` or something similar to avoid confusion